### DISCUSSION TITLE
How to splat an env(<ARRAY>)?

### DISCUSSION THREAD

**Discussion by djthornton1212** [mikefarah/yq#2069] [Q&A] [answered]
I've tried different syntax to splat an env(<ARRAY>) and feel like I'm missing something.

Examples:

Not working 

```bash
list='["item1", "item2", "item3"]' yq --null-input 'env(list)[]'  # returns Error: bad expression, please check expression syntax
list='["item1", "item2", "item3"]' yq --null-input '[env(list)][]'  # nests the array in an array then splats the second level, returning me to where I began
```

Working:

```bash
list='["item1", "item2", "item3"]' yq --null-input 'env(list) | .[]' # pipe then splat
list='["item1", "item2", "item3"]' yq --null-input '[env(list)][][]' # double splat after nesting
```
edit:
I've also found that using the `as` operator allows for correct splatting... is that the word...
```bash
list='["item1", "item2", "item3"]' yq --null-input 'env(list) as $list | $list[]'
```

I guess I'm trying to understand if the `working` examples are the correct way or is there's another method I should be using.

Thanks,

**Comment by mikefarah** [comment]
I just haven't added `env(..)[]` as a thing you can do in the expression parser; which is why it doesn't work. This would be a short hand for `env(list) | .[]`. I'll convert this to a bug and get it fixed. fwiw all the working examples are generated from test code; so they do all work.

**Reply by mikefarah** [reply, marked as answer]
Fixed in v4.44.2

**Reply by djthornton1212** [reply]
Thanks for the response!
